agents/negamax.c

```diff
@@ -0,0 +1,124 @@
+#define MAX_SEARCH_DEPTH 6
+
+#include "negamax.h"
+#include <assert.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "game_logic.h"
```

Comment:
Always put the system headers prior to user-defined ones unless you tend to override the predefined macros. That is,
```
#include <assert.h>
...

#include "game_logic.h"
#include "negamax.h"
```

Do separate with a blank line between the two inclusion areas.